### PR TITLE
Bb data generation

### PR DESCRIPTION
**Pull request by Barisos22**
Added data preparation script, cleared the code for model without hp tuning script, corrected gitignore.

### CODE REVIEW

File: data_preparation.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+from tensorflow.keras.preprocessing.image import ImageDataGenerator
+from sklearn.model_selection import train_test_split
+import keras.models
+import numpy as np
+import os
+from PIL import Image
+import matplotlib.pyplot as plt
+
+#Get the directories of burst and non-burst images
+def get_imgs_directory():
```
Comment (kenfus) [inline data_preparation.py]: No parameters here? What if someone wants to save the data somewhere else?

File: data_preparation.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    nburst_image_dir = os.path.join(c_dir, relative_nburst_dir)
+    
+
+    print(burst_image_dir)
+    print(nburst_image_dir)
+    
+    return burst_image_dir, nburst_image_dir
+
+#resize and fill the images to be the target size: likely 256x256 for possible use in CNN's
+def resize_and_fill(image_path, target_size):
```
Comment (kenfus) [inline data_preparation.py]: I would rather see some keras functions here but if it works, that is OK. However, it's not very controllable. I would rather see keras functions, such as https://www.tensorflow.org/api_docs/python/tf/keras/layers/Resizing

File: data_preparation.py [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+            new_img = resize_and_fill(image_path, target_size)
+            new_img.save(os.path.join(img_directory, file_name)) #save new image
+            
+        
+    plt.imshow(new_img)
+    plt.axis("off")
+    plt.show()
+    
+#Rescale the images
+def rescale(image, label):
```
Comment (kenfus) [inline data_preparation.py]: Same as above, here https://www.tensorflow.org/api_docs/python/tf/keras/preprocessing/image/ImageDataGenerator seems to be nice.

File: data_preparation.py [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+    return image/255. , label
+
+
+#Obtain training and validation datasets from image directories      
+def get_datasets():
```
Comment (kenfus) [inline data_preparation.py]: No parameters? Also image_size is set to 256. Do we then need interpolation from above?

File: data_preparation.py [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+    
+    train_ds = train_ds.map(rescale)
+    validation_ds = validation_ds.map(rescale)
+
+    return train_ds, validation_ds
+
+#Returns the number of batches
+def get_img_num(ds):
+    #Determine the number of images in train and validation datasets
+    ds_total_size = tf.data.experimental.cardinality(ds).numpy()
```
Comment (kenfus) [inline data_preparation.py]: Does this work? Technically, it also tells you when keras creates the dataset above.

**Review by kenfus — CHANGES_REQUESTED** [review]

**Comment by kenfus** [comment]
Much better! 
I think I would like to see in the notebook before you train a model a quick overview of the data with the code I send you from other students:

```
# train dataset
class_names = train_ds.class_names

plt.figure(figsize=(10, 10))
for images, labels in train_ds.take(1):
  for i in range(9):
    ax = plt.subplot(3, 3, i + 1)
    plt.imshow(images[i].numpy().astype("uint8"))
    plt.title(class_names[labels[i]])
    plt.axis("off")
```

Also, I would like to see the model being created in a separate file, very parameterized, so we can start making experiments.

File: data_preparation.py (hunk 1 above)
Comment (Barisos22) [reply data_preparation.py]: I assumed the path the images are stored in from data generation script would be fixed (inside data folder). I can make it dynamic though yes

File: data_preparation.py (hunk 2 above)
Comment (Barisos22) [reply data_preparation.py]: I think using Keras has higher chance of leading to distortions

**Pull request closed without merging**